platforms/software/dependency-management/src/main/java/org/gradle/api/internal/catalog/DefaultVersionCatalogBuilder.java

```diff
@@ -180,7 +179,7 @@ private DefaultVersionCatalog doBuild() {
         maybeImportCatalogs();
         if (!aliasesInProgress.isEmpty()) {
             String alias = aliasesInProgress.iterator().next();
-            throw throwVersionCatalogProblemException(getProblemService().create(builder ->
+            throw throwVersionCatalogProblemException(getProblemService(), getProblemService().forCore().create(builder ->
```

Comment:
🤔 It definitely reads weird to me to have to call a `throwXYZProblemException` method that you have to pass an `InternalProblems` service to, instead of calling that method _on_ the service itself.